base/ltcmdhooks.dtx

```diff
@@ -255,6 +255,27 @@
 % part of the code.\footnote{We might change this behavior in the main
 % document slightly after gaining some usage experience.}
 %
+% \subsection{Command copies}
+%
+% Once a hook has been added to a command, it will be present in any copies
+% of that command which are made. For example, if in the preamble we have
+% \begin{verbatim}
+%   \NewDocumentCommand\foo{}{}
+%   \AddToHook{cmd/foo/after}{1}%
+% \end{verbatim}
+% then after |\begin{document}|, any use of \cs{NewCommandCopy} will
+% include the hook in the copy, for example
+% \begin{verbatim}
+%   \NewCommandCopy\baz\foo
+% \end{verbatim}
+% would include the hook in this copied command.
+%
+% \subsection{Grouping}
+%
+% Command hooks are intended to be added to document commands, which are
+% typically defined globally. As such, adding a hook is a global operation,
+% even if the command was previously only locally-defined. The
+% \emph{content} of hooks respect \TeX{} groupings
```

Comment:
> The _content_ of hooks respect TeX groupings

Still... ambiguous. `\AddToHook` and `\hook_gput_code:nnn` both operate globally.